core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -56,14 +58,30 @@ private[spark] class UnionPartition[T: ClassTag](
   }
 }
 
+object UnionRDD {
```

Comment:
actually, it's best to avoid having a singleton fork join pool here. Just move it into `getPartitions`. It's unlikely that we'll need to share these pools across `UnionRDD`s; each RDD probably has enough partitions to saturate the pool.